content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,310 @@
+# Consensus Algorithms
+
+This document is about explaining, at a high level, how the consensus algorithms work. After reading it, the engineer should understand the mechanism behind the consensus running Flare.
+
+## FBA and Separating Sybil Attack Resistance from Consensus by Making it a Local Selection
+
+### Consensus
+
+A consensus algorithm is an agreement system that follows a predefined set of steps in order to allow a group of participants (nodes) to reach to the same decision about a statement.
+In a blockchain network nodes repeatedly decide what the complete history of a shared ledger is, from among multiple possible versions that occasionally conflict.
+This network-wide agreement allows the recipient of a crypto coin to have faith that the coin is both (a) valid (not counterfeit) and (b) not already spent elsewhere.
+
+### Federated Byzantine Agreement
+
+#### Introduction
+
+Any agreement system in a distributed computing network needs to be fault-tolerant.
+This means it must produce consistent results despite errors like slow communication links, unresponsive nodes and misordered messages.
+A Byzantine agreement system is additionally tolerant of [`Byzantine`](https://medium.com/coinmonks/a-note-from-anthony-if-you-havent-already-please-read-the-article-gaining-clarity-on-key-787989107969) faults: nodes that give false information, whether due to error or in a deliberate attempt to subvert the system or gain some advantage.
```

Comment:
```suggestion
A Byzantine agreement system is additionally tolerant of [`Byzantine faults`](https://medium.com/coinmonks/a-note-from-anthony-if-you-havent-already-please-read-the-article-gaining-clarity-on-key-787989107969): nodes that give false information, whether due to error or in a deliberate attempt to subvert the system or gain some advantage.
```

Also wouldn't this be a better link? https://en.wikipedia.org/wiki/Byzantine_fault

It is also much more likely that a medium article would be deleted/moved/become paywalled, than a Wikipedia article. WDYT?

Reply:
Unfortunately yes, I dont like wikipedia explanations though. They are usually badly explained and from some higher level

Reply:
Corrected and updated the link here https://github.com/optakt/optakt.github.io/pull/21/commits/f524c8d94d32f6edd6da60a9e7f74a848c44d4ce#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R19
Thanks!

Reply:
IMO in this context, high level is good :) Readers are free to dig deeper if they want to, but it's not the purpose of this document. The most important when it comes to external links IMO is that they should be:

* Clear, concise, factual (not tips & tricks-type articles, or blog posts, IMO)
* Reliable to stand the test of time (will not start to 404 because it's a personal blog and the person decided to get rid of it for example)
* From reliable sources (this is where Wikipedia isn't always great, but at least they do peer reviews so it's not perfect but quite reliable I'd say, on average, especially for non-political/historical subjects)

Reply:
Yes, what I said is that often Wikipedia links are not clear, concise and factual